ckan/public/base/javascript/modules/image-upload.js

```diff
@@ -204,6 +204,15 @@ this.ckan.module('image-upload', function($) {
      */
     _onInputChange: function() {
       var file_name = this.input.val().split(/^C:\\fakepath\\/).pop();
+
+      // Internet Explorer 6-11 and Edge 20+
+      var isIE = false || !!document.documentMode;
+      var isEdge = !isIE && !!window.StyleMedia;
+      // for IE/Edge when 'include filepath option' is enabled
+      if (isIE || isEdge) {
+        (file_name=file_name.match(/[^\\\/]+$/)) && file_name[0];
```

Comment:
* Second part of conjunction does nothing;
* even though match object will be converted to string bellow, in `val` method, it would be better to make things explicit and avoid potential bugs if the surrounding code will be modified.